gateway/ln-gateway/src/rpc/rpc_server.rs

```diff
@@ -26,9 +26,34 @@ pub async fn run_webserver(
     gateway: Gateway,
     task_group: &mut TaskGroup,
 ) -> axum::response::Result<()> {
-    let (routes, admin_routes) = if let Some(gateway_config) = config {
+    let v1_routes = v1_routes(config, gateway.clone());
+    let api_v1 = Router::new()
+        .nest(&format!("/{V1_API_ENDPOINT}"), v1_routes.clone())
+        // Backwards compatibility: Continue supporting gateway APIs without versioning
+        .merge(v1_routes);
```

Comment:
Should we add an API endpoint that tells us the min/max supported version?

Reply:
Not something commonly seen. I guess it's because it falls out of versioned apis, rendering it weirdly special.

In web apis apps usually target a single api version and it's server side job to keep around older versions long enough. Unclear if the same strategy is OK here. But the app can just iterate and try couple of versions to discover which one is available if really needed anyway.

Reply:
following up with this endpoint

Reply:
The problem is we want newer clients to be compatible with older federations, that's not a problem centralized APIs typically have. I think we want backwards and forwards compatibility strictly speaking. For that having some discover mechanism (either iteration or an endpoint) is very useful.

Reply:
I want to read more on [OpenAPI v3.1.0](https://spec.openapis.org/oas/latest.html#openapi-object) spec which defines an info object with a version field. We can aim for spec compliance, serve the api docs at a path like `/` or `/openapi` or `/api-docs` and then expect clients to use this for discovering api capabilities

Reply:
I doubt adhering to a spec which to read and understand would take multiple days is worth the effort unless there is really good rust tooling already that removes this necessity.